src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -2892,10 +2894,7 @@ export default class EthereumApi implements Api {
     const fromString = tx.from.toString();
 
     const wallet = this.#wallet;
-    const encryptedKeyFile = wallet.encryptedKeyFiles.get(fromString);
-    if (encryptedKeyFile === undefined) {
-      throw new Error("no key for given address or file");
-    }
+    const secretKey = await wallet.getFromKeyFile(fromString, passphrase);
```

Comment:
Previously, we would try to get the `fromString` out of `encryptedKeyFiles`. If the result was `undefined`, we'd throw, but if the result was `null`, we'd queue the transaction without the secret key.

I couldn't find a code path that would lead to the result of `encryptedKeyFiles.get(fromString)` being `null`, only undefined. So, the `getFromKeyFile` function throws for both `null` and `undefined`.

If we can find a case where `null` is a possible/valid result, we might need to back out this change.